### PR TITLE
[Refactor] Refactor Registration and Improve Performance of SPIN

### PR DESCRIPTION
**Pull request by mingyuan-zhang**
Modification Log:

1. Fix bugs of camera parameters in registration
2. Fix bugs of `sample_idx` in `fits_dict`
3. Add cache to SPIN training
4. Refactor registration configuration in `mesh_estimator` to support using `static_fit` only
5. Refactor model optimization
6. Add `quiet` parameter to `simplify`

**Accuracy Comparison**
|  | MPJPE | PA-MPJPE |
|:------:|:-------:|:------:|
| Old | 96.06 | 59.06 |
| New | **94.11** | **57.54** |

### CODE REVIEW

File: configs/spin/resnet50_spin_pw3d.py [hunk 1]
```diff
@@ -57,7 +57,10 @@
         in_ndc=False,
         focal_length=5000,
         image_size=(img_res, img_res),
-        principal_point=(img_res / 2, img_res / 2)))
+        principal_point=(img_res / 2, img_res / 2)),
+    quiet=True)
```
Comment (caizhongang) [inline configs/spin/resnet50_spin_pw3d.py:61]: Let's use `verbose` as the opposite of `quiet`.

File: configs/spin/resnet50_spin_pw3d.py [hunk 2]
```diff
@@ -158,41 +170,47 @@
                 data_prefix='data',
                 pipeline=train_pipeline,
                 convention='smpl_49',
+                cache_data_path=cache_files['h36m'],
```
Comment (caizhongang) [inline configs/spin/resnet50_spin_pw3d.py:173]: Shall we indicate `cache` in the config name?

File: mmhuman3d/models/architectures/mesh_estimator.py [hunk 3]
```diff
@@ -248,13 +240,9 @@ def run_registration(
...
         gt_out = self.body_model_train(
             betas=gt_betas, body_pose=gt_pose, global_orient=gt_global_orient)
-        if num_keypoints == 49:
-            gt_model_joints = gt_out['joints']
-            gt_vertices = gt_out['vertices']
-        else:
-            gt_model_joints = gt_out['joints'][:, 25:, :]
-            gt_vertices = gt_out['vertices']
-        # TODO: add joint mask
+        assert num_keypoints == 49
```
Comment (caizhongang) [inline mmhuman3d/models/architectures/mesh_estimator.py:243]: Let's leave a `TODO` comment to indicate future modification that allows for more keypoint conventions.

File: mmhuman3d/models/registrants/smplify.py [hunk 4]
```diff
@@ -71,6 +71,7 @@ def __init__(
         ignore_keypoints: List[int] = None,
         device=torch.device('cuda' if torch.cuda.is_available() else 'cpu'),
         verbose: bool = False,
+        quiet: bool = False,
```
Comment (caizhongang) [inline mmhuman3d/models/registrants/smplify.py:74]: We should combine `verbose` and `quite` into one.

File: configs/spin/resnet50_spin_pw3d.py (hunk 1 above)
Comment (mingyuan-zhang) [reply configs/spin/resnet50_spin_pw3d.py]: done

File: mmhuman3d/models/architectures/mesh_estimator.py (hunk 3 above)
Comment (mingyuan-zhang) [reply mmhuman3d/models/architectures/mesh_estimator.py:246]: Done

**Review by caizhongang — APPROVED** [review]

**Pull request merged**